chatmeleon-server/prisma/schema.prisma

```diff
@@ -87,3 +89,15 @@ model Message {
     senderId String @db.ObjectId
     sender   User   @relation(fields: [senderId], references: [id], onDelete: Cascade)
 }
+
+// Define the Friendship model
+model User_Relationship {
+    id        String   @id @default(auto()) @map("_id") @db.ObjectId
+    type      String
```

Comment:
Is there no other way of defining enums in Prisma? Arbitrary string value type can easily be a hard to debug source of error later on.

Reply:
Well the ID is by default mapped to a unique identifier generated by MongoDB (as u can see in the code), and yea Prisma does not support enums. But MongoDb itself does have configurations for generating unique identifiers so maybe this can be a topic for another day but its worth noting tho

Reply:
I meant for the `type String`. In our implementation, we're basically using this field for either values of `pending_friend_request`, `sending_friend_request` or `friend` and whatnot. What I'm saying is, using just raw strings like this can create situations where one of the dev mistype the value of the string somewhere deep in the code base (e.g., `pending_friend_requests`) and that would be a really hard to debug point of misbehavior since nothing validates this value used, so the only way to debug then would be to manually check for each line of usage of this field.

Reply:
We can discuss this another time or would be great if someone knows or can find a solution to this now

Reply:
Ok so this Thursday then lol